### DISCUSSION TITLE
[Question] How to make Bull jobs concurrent

### DISCUSSION THREAD

**Discussion by thakee-kochasoft** [OptimalBits/bull#2463] [Q&A] [answered]
## Description
For the context, I'm new to Bull and trying to execute a lot of jobs concurrently. But Bull seems like it'll wait for one job to complete.

## Minimal, Working Test code to reproduce the issue.
```js
const Bull = require("bull");


const jobs = new Bull('jobs');


jobs.process(async (job) => {
  const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
  console.log(`start job.id = ${job.id}`);
  await wait(3000); // Wait 3 seconds.
  console.log(`end   job.id = ${job.id}`);
});


void async function main () {
  for (let i = 0; i < 10; i++) {
    jobs.add({});
  }
}();
```

If I ran this code, it'll take 30 seconds to execute, one job after another. What am I doing wrong and how to make this concurrent?

## Bull version
4.10.1

**Comment by rbartoli** [comment]
```
// You can use concurrency as well:
queue.process(5, '/path/to/my/processor.js');
```

From the docs on https://github.com/OptimalBits/bull

**Comment by manast** [comment, marked as answer]
Also if you are new to Bull you probably should start with BullMQ instead: https://docs.bullmq.io
